elyra/pipeline/kfp/kfp_metadata.py

```diff
@@ -33,14 +33,23 @@ def on_load(self, **kwargs: Any) -> None:
             # Inject auth_type property for metadata persisted using Elyra < 3.3:
             # - api_username and api_password present -> use DEX Legacy
             # - otherwise -> use no authentication type
-            if self.metadata.get('api_username') is None or\
-               len(self.metadata.get('api_username').strip()) == 0 or\
-               self.metadata.get('api_password') is None or\
-               len(self.metadata.get('api_password').strip()) == 0:
+            if len(self.metadata.get('api_username', '').strip()) == 0 or\
```

Comment:
Note: this simplifying change (this line and next) is unrelated to the goal of this PR but was included since the file is touched by other changes.